### PR TITLE
Docker documentation

### PR DESCRIPTION
**Pull request by rbuysse**

### CODE REVIEW

File: docs/source/sawtooth_developers_guide/sawtooth_docker.rst [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ 
+sawtooth-dev-\*
+  Dockerfiles used for testing and development. 
+
+  These Dockerfiles are built as part of the `build_all` command and run as part
+  of the `run_tests` command.
+ 
+sawtooth-\*
+  Dockerfiles that build images with Sawtooth Lake installed from the Sawtooth
+  Lake package repository (repo.sawtooth.me)[repo.sawtooth.me]. These Dockerfiles
```
Comment (aludvik) [inline docs/source/sawtooth_developers_guide/sawtooth_docker.rst]: I messed up the markdown formatting for the repo.sawtooth.me link... Will change it on mine but it should be changed here too.

File: docs/source/tutorial.rst [hunk 2]
```diff
@@ -414,12 +512,18 @@ tree, using the sawtooth CLI.
...
-Log in to the Vagrant environment with the command `vagrant ssh` to run the commands below. 
+Log in to the Vagrant environment with the command `vagrant ssh` to run the commands below.
 
 
 Starting the Rest API
 ---------------------
 
+.. caution::
+
+  As with the transaction processors above, Rest API transaction processor is
```
Comment (aludvik) [inline docs/source/tutorial.rst]: Is the Rest API considered a transaction processor? @delventhalz

File: docs/source/sawtooth_developers_guide/sawtooth_docker.rst [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  sawtooth-validator
+
+
+compose/
+  Contains docker compose files for easily bringing up validator networks and
+  connected transaction processors.
+
+sawtooth-build-\*
+  Dockerfiles used to build the Sawtooth Lake source code without polluting the
+  host environment. The containers should be run with the source code mounted
```
Comment (ojalatodd) [inline docs/source/sawtooth_developers_guide/sawtooth_docker.rst]: Could you elaborate here? "Should" is prescriptive.   Is the source code automatically "mounted inside the container"? The phrasing makes it sound like the user needs to do something. Or is this a warning to NOT do something? Explain.

File: docs/source/sawtooth_developers_guide/sawtooth_docker.rst [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+sawtooth-dev-\*
+  Dockerfiles used for testing and development. 
+
+  These Dockerfiles are built as part of the `build_all` command and run as part
+  of the `run_tests` command.
+ 
+sawtooth-\*
+  Dockerfiles that build images with Sawtooth Lake installed from the Sawtooth
+  Lake package repository (http://repo.sawtooth.me). These Dockerfiles
+  do not make any assumptions about the build or runtime environments.
```
Comment (ojalatodd) [inline docs/source/sawtooth_developers_guide/sawtooth_docker.rst]: Do the other Dockerfiles make assumptions about the build or runtime environments?

File: docs/source/tutorial.rst [hunk 5]
```diff
@@ -72,9 +59,93 @@ Open up a terminal and run the following:
...
+
+Your environment is ready! Continue on to `Multi-language support for transaction processors`_.
+
+Resetting The Environment
+-------------------------
+
+If the environment needs to be reset for any reason, it can be returned to
+the default state by logging out of the client container, then pressing
+CTRL-c from the window where you originally ran docker-compose. Once the
+containers have all shut down:
```
Comment (ojalatodd) [inline docs/source/tutorial.rst]: Do what? Suggestion:
Once the containers have all shut down, enter the command 'docker-compose -f sawtooth-demo.yaml down':

**Pull request merged**